src/classlibnative/bcltype/number.cpp

```diff
@@ -137,86 +137,185 @@ unsigned int Int64DivMod1E9(unsigned __int64* value)
 
 void ecvt( double value, int count, int* dec, int* sign, wchar_t* digits )
 {
+    // ========================================================================================================================================
+    // This implementation is based on the paper: https://www.cs.indiana.edu/~dyb/pubs/FP-Printing-PLDI96.pdf
+    // Besides the paper, some of the code and ideas are modified from http://www.ryanjuckett.com/programming/printing-floating-point-numbers/
```

Comment:
not positive, but I'm fairly certain we need to preserve the license header listed by the site.
@tarekgh can confirm (and follow up with LCA if necessary).